### DISCUSSION TITLE
Filter Products in Collection by Metafield Value

### DISCUSSION THREAD

**Discussion by grichards** [Shopify/storefront-api-feedback#122] [Ideas & Requests]
[As referenced here](https://shopify.dev/themes/navigation-search/filtering/storefront-filtering) it is possible to filter a Collection by Metafield values if you are using a traditional Liquid Storefront.

[As seen here](https://shopify.dev/api/examples/filter-products) there is no reference to how to achieve the same kind of filtration using GraphQL.

I want to create filters for my Hydrogen store using Metafield Definitions and the Metafield values associated with Products within a given Collection.... exactly the same as can be done using an Online Store 2.0 Liquid Theme.

Am I simply not finding the documentation... or is this indeed a missing feature of Shopify GraphQL and Hydrogen?

**Comment by johncraigcole** [comment]
Hey @grichards I know we more or less got this squared away [here](https://github.com/Shopify/storefront-api-feedback/discussions/72), I just wanted to follow up that I added an example to our [docs](https://shopify.dev/api/examples/filter-products#query-products-by-metafield-value).

**Reply by grichards** [reply]
Hi @johncraigcole . Yes thank you!

While I have your attention, may I inquire about [this](https://github.com/Shopify/storefront-api-feedback/discussions/123).

Is the idea / request / value clear?

Any chance a feature like this is in Shopify's roadmap?

TY!
G
